exporter/syslogexporter/config.go

```diff
@@ -56,7 +57,8 @@ func (cfg *Config) Validate() error {
 		invalidFields = append(invalidFields, errInvalidEndpoint)
 	}
 
-	if strings.ToLower(cfg.Network) != "tcp" && strings.ToLower(cfg.Network) != "udp" {
+	cfg.Network = strings.ToLower(cfg.Network)
```

Comment:
It may be possible to make `Network` a `confignet.TransportType` instead of a string, then you could directly compare `Network` and `confignet.TransportTypeTCP`. `confignet.TransportType` has an unmarshal function so it should be a non-breaking change for end users.

Reply:
If I see correctly, [UnmarshalText](https://github.com/open-telemetry/opentelemetry-collector/blob/dc48d0e66b39dc5a0cd2bac68d964eef70863219/config/confignet/confignet.go#L35) accepts only "tcp" and "udp" and doesn't accept "TCP" and "UDP", current version of syslog export accepts all options.

Reply:
You're correct.  If the exporter supports `TCP` and `UDP` then I agree that sticking with the string is appropriate.